usaspending_api/submissions/helpers.py

```diff
@@ -132,8 +131,12 @@ def get_last_closed_periods_per_year() -> List[ClosedPeriod]:
                             submission_reveal_date <= now()
                     order   by submission_fiscal_year, -submission_fiscal_month
                 ) as m on m.submission_fiscal_year = q.submission_fiscal_year
+        where   q.submission_fiscal_year = %(fiscal_year)s
```

Comment:
I'm concerned about this query a bit. It seems a bit too smart for its own good. 

In the case where it is just monthly period 2 or monthly periods 2 and 3 that have closed, but the first quarer, Q1, has not yet closed, this query returns no results. 
However it should somehow get the latest monthly period that has closed, yes?

Run these statements in sequence in sandbox to see what I mean (don't forget to rollback!)
```sql
begin;
update dabs_submission_window_schedule set submission_reveal_date = '9999-12-31 23:59:59.999999+00' where id > 2021030;
select * from dabs_submission_window_schedule order by submission_reveal_date desc, id desc;
select  coalesce(q.submission_fiscal_year, m.submission_fiscal_year) as fiscal_year,
        q.submission_fiscal_quarter as fiscal_quarter,
        m.submission_fiscal_month as fiscal_month
from    (
            select  distinct on (submission_fiscal_year)
                    submission_fiscal_year, submission_fiscal_quarter
            from    dabs_submission_window_schedule
            where   is_quarter is true and
                    submission_reveal_date <= now()
            order   by submission_fiscal_year, -submission_fiscal_quarter
        ) as q
        full outer join (
            select  distinct on (submission_fiscal_year)
                    submission_fiscal_year, submission_fiscal_month
            from    dabs_submission_window_schedule
            where   is_quarter is false and
                    submission_reveal_date <= now()
            order   by submission_fiscal_year, -submission_fiscal_month
        ) as m on m.submission_fiscal_year = q.submission_fiscal_year
where   q.submission_fiscal_year = 2021
;
rollback;
```

What it used to return before the FY where clause added
![image](https://user-images.githubusercontent.com/14953413/122050912-fc41ac80-cdb1-11eb-8cc2-20182f4814ae.png)

But what it returns in the case of the above data and query:
![image](https://user-images.githubusercontent.com/14953413/122050970-0fed1300-cdb2-11eb-8cda-bd127ded40ec.png)

I think the `WHERE` needs to be moved outside the full outer join, as applied on the full results of the query as if it were a subquery. 

But then I worry about other times of the year and if that join on `m.submission_fiscal_year = q.submission_fiscal_year` might be incorrectly joining quarterly and monthly data. That may be ok though -- needs further tests at various times during the year (like when P12 has closed but no yet Q4...).

Reply:
Seems to work okay as long as you apply the year filtering as an outer-query. Alternatively, you could just get all years, and then filter the handful of results in python, to pick out the year you're interested in.

Reply:
After looking closer I don't think that SQL is needed at all. New change removes that function and creates a ClosedPeriod object with the filters provided. This is closer to the previous functionality but was overlooked during initial changes. The main difference between old vs new is that the new code is not going through all closed periods since they really don't matter, only the period we are searching for.